### PR TITLE
107 move notification configs from a modal to a page

### LINKED ISSUE #107: Move NotificationConfigs from a modal to a page...
## Is your feature request related to a problem? Please describe

Having things in a modal feels a bit wrong from a UX perspective, particularly when we have modals to select recipeinets etc.

## Describe the solution you'd like

Similar to recipient lists where you navigate to a page like http://localhost:3007/recipients/lists/302bb909-3098-4011-9754-32e844c7f224

Would allow us to use tabs if needed for different configuration portions if needed e.g. selecting SQL queries in a tab?

### PR DESCRIPTION
**Pull request by jmbrunskill**
Closes #107 

# 👩🏻‍💻 What does this PR do?
 

Moves to using a actual page rather than modal to manage notification configurations, which should give more flexibility and allow PRs such as #116 to add more features such as parameters.

The ui is modeled of open mSupply with a app bar and app footer design.

image

# 🧪 How has/should this change been tested?

Right now only the cold chain config is in this branch. Similar config will need to be updated for the scheduled reports and any other future configuration types.

## 💌 Any notes for the reviewer?

I'm not sure if the generics are actually required here, but tried not to change things too much from what Lache had done with modals.

### CODE REVIEW

File: frontend/packages/system/src/Notifications/ListView/ListView.tsx [hunk 1]
```diff
@@ -13,16 +13,36 @@ import {
   createTableStore,
   useColumns,
 } from '@common/ui';
-import { NotificationsModal } from '../Modals/NotificationsModal';
 import { useEditModal, useQueryParamsState } from '@common/hooks';
 import { NotificationConfigRowFragment, useNotificationConfigs } from '../api';
 import { useDeleteNotificationConfig } from '../api/hooks/useDeleteNotificationConfig';
+import { ConfigKind, useNavigate } from 'packages/common/src';
```
Comment (mark-prins) [inline frontend/packages/system/src/Notifications/ListView/ListView.tsx:19]: importing directly from `packages/` like this is fine in the notify app - but breaks tests in open mSupply 🤷 
It makes this suggestion optional - though the import is neater looking if you use the package name:
```suggestion
import { ConfigKind, useNavigate } from '@notify-frontend/common';
```

File: frontend/packages/system/src/Notifications/ListView/ListView.tsx [hunk 2]
```diff
@@ -13,16 +13,36 @@ import {
...
+import { configRoute } from '../navigate';
+import { CreateConfigKindModal } from '../Pages/CreateConfigKindModal';
```
Comment (mark-prins) [inline frontend/packages/system/src/Notifications/ListView/ListView.tsx:21]: I found the name misleading - the button is for a `New Notification` : the modal sounds like it's creating a `ConfigKind`. Perhaps `NewNotificationModal` or `CreateNotificationModal` 🤷

File: frontend/packages/system/src/Notifications/Pages/Base/BaseNotificationEditPage.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                height={64}
+              >
+                <Box flex={1} display="flex" justifyContent="flex-end" gap={2}>
+                  <ButtonWithIcon
+                    shrinkThreshold="lg"
+                    Icon={<CloseIcon />}
+                    label={t('button.close')}
+                    color="secondary"
+                    sx={{ fontSize: '12px' }}
+                    onClick={() => navigateUpOne()}
```
Comment (mark-prins) [inline frontend/packages/system/src/Notifications/Pages/Base/BaseNotificationEditPage.tsx:100]: no need to create a new function here
```suggestion
                    onClick={navigateUpOne}
```

File: frontend/packages/system/src/Notifications/Pages/CreateConfigKindModal.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+interface SelectNotificationTypeModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export const CreateConfigKindModal = ({
+  isOpen,
+  onClose,
+}: SelectNotificationTypeModalProps) => {
+  const t = useTranslation(['system']);
```
Comment (mark-prins) [inline frontend/packages/system/src/Notifications/Pages/CreateConfigKindModal.tsx:27]:
```suggestion
  const t = useTranslation('system');
```

File: frontend/packages/system/src/Notifications/Pages/CreateConfigKindModal.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+                    title: title,
+                  },
+                }).then(() => {
+                  navigate(configRoute(kind, id));
+                });
+              }
+            }}
+            isLoading={isLoading}
+            startIcon={<ArrowRightIcon />}
+            variant="contained"
```
Comment (mark-prins) [inline frontend/packages/system/src/Notifications/Pages/CreateConfigKindModal.tsx:61]: not required - the BaseButton is also setting the variant to `contained`

**Review by mark-prins — APPROVED** [review]
👍 Very nicely done!

I have a few layout suggestions, which I've added to a [new branch](https://github.com/openmsupply/notify/compare/107-move-notificationconfigs-from-a-modal-to-a-page...107-move-notification-configs-suggestion) as they touch a few files.

I've added some spacing to this modal:

Screenshot 2023-09-15 at 12 44 08 PM

like this:
image

And thought that the recipient list would be better with a label:

Screenshot 2023-09-15 at 1 36 19 PM

there's a few other comments, all minor bits tho

File: frontend/packages/system/src/Notifications/ListView/ListView.tsx (hunk 1 above)
Comment (jmbrunskill) [reply frontend/packages/system/src/Notifications/ListView/ListView.tsx]: I rely too much on the auto importer :(
Thanks for picking this up!

**Pull request merged**